.circleci/config.yml

```diff
@@ -1,22 +1,35 @@
 version: 2.1
+
+orbs:
+  shellcheck: circleci/shellcheck@1.3.15
+
 executors:
   circleci-gauge-executor:
     docker:
       - image: circleci/node:11.9.0-browsers
 
+workflows:
+  version: 2
+  test:
+    jobs:
+      - shellcheck/check
+      - test:
+          requires:
+            - shellcheck/check
+
 jobs:
   test:
     working_directory: ~/agile-pathway-co-uk
     executor: circleci-gauge-executor
     steps:
       - checkout
       - run: curl -SsL https://downloads.gauge.org/stable | sh
+      - run:
+          name: Define TEST_SITE_URL Environment Variable at Runtime
+          command: ./.circleci/scripts/set-test-site-url.sh
+      - run:
+          name: Check test site is up
+          command: ./.circleci/scripts/check-test-site-up.sh
       - run: cd e2etests && gauge install && npm install && npm test
       - store_artifacts:
           path: reports
-
-workflows:
-  version: 2
-  build-test:
-    jobs:
-      - test
```

Comment:
The CircleCI config is not optimised yet (in terms of caching, separation of concerns etc) - that can come later, [if/when needed](https://wiki.c2.com/?PrematureOptimization).